File: examples/python/02_python_basic_api/students_test.py
# Copyright 2021 Touca, Inc. Subject to Apache-2.0 License.

import touca
from students import parse_profile, calculate_gpa


@touca.Workflow
def test_students(testcase: str):
    student = parse_profile(testcase)
    touca.add_assertion("username", student.username)
    touca.add_result("fullname", student.fullname)
    touca.add_result("birth_date", student.dob)
    touca.add_result("gpa", calculate_gpa(student.courses))


if __name__ == "__main__":
    touca.run()

File: sdk/java/examples/03_java_core_api/src/main/java/io/touca/examples/core/StudentsTest.java
// Copyright 2021 Touca, Inc. Subject to Apache-2.0 License.

package io.touca.examples.core;

import io.touca.Touca;

public final class StudentsTest {
    public static void main(String[] args) {
        Touca.configure(options -> {
            options.offline = true;
        });
        if (!Touca.isConfigured()) {
            System.err.print(Touca.configurationError());
            System.exit(1);
        }
        for (String username : Touca.getTestCases()) {
            Touca.declareTestcase(username);
            Touca.startTimer("parse_profile");
            Student student = Students.parseProfile(username);
            Touca.stopTimer("parse_profile");

            Touca.addSerializer(Course.class, course -> {
                return course.name;
            });
            Touca.addAssertion("username", student.username);
            Touca.addResult("fullname", student.fullname);
            Touca.addResult("birth_date", student.dob);

            Touca.scopedTimer("calculate_gpa", () -> {
                Touca.addResult("gpa", Students.calculateGPA(student.courses));
            });
            Touca.addMetric("external_source", 1500);

            Touca.post();
            Touca.saveJson(String.format("touca_%s.json", username));
            Touca.saveBinary(String.format("touca_%s.bin", username));
            Touca.forgetTestcase(username);
        }
        Touca.seal();
    }
}
